index.js

```diff
@@ -282,26 +282,16 @@ module.exports = class Hypercore extends EventEmitter {
     if (!isFirst) await opts._opening
     if (opts.preload) opts = { ...opts, ...(await this._retryPreload(opts.preload)) }
 
-    const keyPair = (key && opts.keyPair)
-      ? { ...opts.keyPair, publicKey: key }
-      : key
-        ? { publicKey: key, secretKey: null }
-        : opts.keyPair
-
-    // This only works if the hypercore was fully loaded,
-    // but we only do this to validate the keypair to help catch bugs so yolo
-    if (this.key && keyPair) keyPair.publicKey = this.key
+    const keyPair = opts.keyPair
 
     if (opts.auth) {
       this.auth = opts.auth
-    } else if (opts.sign) {
-      this.auth = Core.createAuth(this.crypto, keyPair, opts)
```

Comment:
will anything break if we remove this branch?

Reply:
we tested and corestore doesn't use it so doubt it

Reply:
yeh think corestore just passes `auth` so no problem there

Reply:
👍